pycloudlib/azure/cloud.py

```diff
@@ -665,6 +669,13 @@ def _create_virtual_machine(
         except HttpResponseError as e:
             err_code = e.error.code
             err_msg = e.error.message
+            if (
+                err_code in ("OperationNotAllowed", "QuotaExceeded")
+                and "quota" in err_msg.lower()
+            ):
+                if "limit" in err_msg.lower():
+                    raise QuotaLimitError(err_msg) from e
+                raise InsufficientQuotaError(err_msg) from e
```

Comment:
add some inline comments explaining further. Also add "Raises:" section to docstring.